packages/commerce-server/src/@types/index.ts

```diff
@@ -60,13 +60,14 @@ export type SanityProduct = {
   name: string
   title?: string
   action: string
+  type?: 'self-paced' | 'live'
   description?: string
   slug?: string
   image: {
     url: string
     alt: string
   }
-  state?: 'draft' | 'active' | 'unavailable'
+  state?: 'draft' | 'active' | 'unavailable' | 'archived'
   summary?: PortableTextBlock
   modules: SanityProductModule[]
   upgradableTo?: SanityProduct
```

Comment:
added new `type` type. we were also missing `archived` state.

Reply:
we've got a separate `Product` type that actually has a schema in the commerce package that I prefer generally because it has a backing schema where this one is pre-Zod